Model/Resolver/GAResolver.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Elgentos\ServerSideAnalytics\Model\Resolver;
+
+use Elgentos\ServerSideAnalytics\Model\ResourceModel\SalesOrder\CollectionFactory;
+use Elgentos\ServerSideAnalytics\Model\SalesOrderRepository;
+use Exception;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Quote\Model\MaskedQuoteIdToQuoteIdInterface;
+
+class GAResolver implements ResolverInterface
+{
+    public function __construct(
+        protected CartRepositoryInterface $quoteRepository,
+        protected MaskedQuoteIdToQuoteIdInterface $maskedQuoteIdToQuoteId,
+        protected CollectionFactory $elgentosSalesOrderCollectionFactory,
+        protected SalesOrderRepository $elgentosSalesOrderRepository,
+    ) {
+        $this->quoteRepository = $quoteRepository;
+        $this->maskedQuoteIdToQuoteId = $maskedQuoteIdToQuoteId;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (is_numeric($args['cartId']) && false === $context->getExtensionAttributes()->getIsCustomer()) {
+            throw new GraphQlAuthorizationException(__('The current customer isn\'t authorized.'));
+        }
+
+        $cartId = is_numeric($args['cartId'])
+            ? $args['cartId']
+            : $this->maskedQuoteIdToQuoteId->execute($args['cartId']);
+
+        /** @var Collection $elgentosSalesOrder */
+        $elgentosSalesOrderCollection = $this->elgentosSalesOrderCollectionFactory->create();
+        $elgentosSalesOrder = $elgentosSalesOrderCollection
+            ->addFieldToFilter('quote_id', $cartId)
+            ->getFirstItem();
```

Comment:
getFirstItem will create a new salesOrder model if it has not been found in the database
https://github.com/magento/magento2/blob/c22a9db44a704b892b81746dc5f707fd87f6e5bf/lib/internal/Magento/Framework/Data/Collection.php#L305